packages/trpc/server/routers/viewer/bookings/get.handler.ts

```diff
@@ -286,85 +285,111 @@ export async function getBookings({
+        },
+        userId: user.id,
+      },
+    }),
+
+    // Retrieve bookings where the user is an attendee
+    prisma.attendee.findMany({
+      where: { email: user.email },
+      select: { bookingId: true },
+    }),
+    // Retrieve team bookings for team-admins and team-owners
+    prisma.membership.findMany({
+      where: { userId: user.id, role: { in: ["ADMIN", "OWNER"] }, accepted: true },
+      select: {
+        team: {
+          select: {
+            eventTypes: {
+              select: {
+                children: {
+                  select: { bookings: { select: { id: true } } },
+                },
+                bookings: { select: { id: true } },
               },
             },
           },
-        ],
-        AND: [passedBookingsStatusFilter, ...filtersCombined],
+        },
       },
-      orderBy,
-      take: take + 1,
-      skip,
     }),
-    prisma.booking.findMany({
-      where: {
-        OR: [
-          {
-            eventType: {
-              team: {
-                members: {
-                  some: {
-                    userId: user.id,
-                    role: {
-                      in: ["ADMIN", "OWNER"],
-                    },
+
+    // Retrieve all sub-teams team bookings for org-admins and org-owners
+    prisma.membership.findMany({
+      where: { userId: user.id, role: { in: ["ADMIN", "OWNER"] }, team: { isOrganization: true } },
+      select: {
+        team: {
+          select: {
+            children: {
+              select: {
+                eventTypes: {
+                  select: {
+                    children: { select: { bookings: { select: { id: true } } } },
+                    bookings: { select: { id: true } },
                   },
                 },
               },
             },
           },
-        ],
-        AND: [passedBookingsStatusFilter, ...filtersCombined],
+        },
       },
-      orderBy,
-      take: take + 1,
-      skip,
     }),
-    prisma.booking.findMany({
-      where: {
-        OR: [
-          {
-            user: {
-              teams: {
-                some: {
-                  team: {
-                    isOrganization: true,
-                    members: {
-                      some: {
-                        userId: user.id,
-                        role: {
-                          in: ["ADMIN", "OWNER"],
+
+    // Retrieve all team users personal bookings for org-admins and org-owners
+    prisma.membership.findMany({
+      where: { userId: user.id, role: { in: ["ADMIN", "OWNER"] }, team: { isOrganization: true } },
+      select: {
+        team: {
+          select: {
+            children: {
+              select: {
+                members: {
+                  select: {
+                    user: {
+                      select: {
+                        eventTypes: {
+                          select: {
+                            bookings: { select: { id: true } },
+                          },
```

Comment:
This query could be very slow on prod. we would have to test this query on big dataset